File: SubAgenticAI.Core/MissionStatus.cs
namespace SubAgenticAI.Core;

/// <summary>
/// Represents the lifecycle status of a SubAgent mission
/// </summary>
public enum MissionStatus
{
    /// <summary>
    /// Mission has been created but not yet started
    /// </summary>
    Created,
    
    /// <summary>
    /// Mission is currently active
    /// </summary>
    Active,
    
    /// <summary>
    /// Mission is paused/suspended
    /// </summary>
    Paused,
    
    /// <summary>
    /// Mission completed successfully
    /// </summary>
    Completed,
    
    /// <summary>
    /// Mission failed or was aborted
    /// </summary>
    Failed,
    
    /// <summary>
    /// Mission has been retired/disposed
    /// </summary>
    Retired
}

File: subagentic_ai_py/subagentic_ai/mission_status.py
"""Mission status enumeration for SubAgent lifecycle management."""

from enum import Enum


class MissionStatus(Enum):
    """Represents the lifecycle status of a SubAgent mission."""
    
    CREATED = "created"  # Mission has been created but not yet started
    ACTIVE = "active"  # Mission is currently active
    PAUSED = "paused"  # Mission is paused/suspended
    COMPLETED = "completed"  # Mission completed successfully
    FAILED = "failed"  # Mission failed or was aborted
    RETIRED = "retired"  # Mission has been retired/disposed
